apps/photos/src/pages/gallery/index.tsx

```diff
@@ -344,7 +344,7 @@ export default function Gallery() {
         }
         const href = `/gallery${collectionURL}`;
         router.push(href, undefined, { shallow: true });
-    }, [activeCollection]);
+    }, []);
```

Comment:
why this ?

Reply:
Error: Cancel rendering route: for this Error, Tried removing activeCollection, error was handled , It doesn't seem have affected any other navigations also

Reply:
But this `useEffect` won't work without the `activeCollection` dependency no? We need to update the navbar URL when the `collectionID` changes. Can you check if there is an option to check if router render is in progress and await on that if possible and then update the route

Reply:
setTimeout(() => {
    const href = `/gallery${collectionURL}`;
    router.push(href, undefined, { shallow: true });
  }, 0); Will setting such a timeout work?

Reply:
How will this help. How have you seen some issues where people have suggested this?

Reply:
When setting a timeout, the issue seems to have resolved, Not sure of it's side effects